lib/brains/tests/test_webhook.py

```diff
@@ -109,14 +109,83 @@ def setUp(self):
             seller_product=self.seller_product,
             provider_id='some-bt:id'
         )
-        self.sub = subscription
+        self.sub = subscription()
+
+
+class TestTransactions(SubscriptionTest):
+
+    def process(self, subscription):
+        process = Processor(notification(subject=subscription))
+        process.update_transactions(process.webhook.subscription,
+                                    self.braintree_sub)
+
+    def test_ignored(self):
+        sub = subscription(transactions=[transaction(status='settling')])
+        self.process(sub)
+        eq_(Transaction.objects.count(), 0)
+
+    def test_caught(self):
+        sub = subscription(transactions=[transaction(status='settled')])
+        self.process(sub)
+        trans = Transaction.objects.get()
+        eq_(trans.status, constants.STATUS_CHECKED)
+
+    def test_processor_declined(self):
+        sub = subscription(transactions=[
+            transaction(status='processor_declined',
+                        processor_response_code='wat'),
+        ])
+        self.process(sub)
+        trans = Transaction.objects.get()
+        eq_(trans.status, constants.STATUS_FAILED)
+        eq_(trans.status_reason, 'processor_declined wat')
+
+    def test_settlement_declined(self):
+        sub = subscription(transactions=[
+            transaction(status='settlement_declined',
+                        processor_settlement_response_code='wat'),
+        ])
+        self.process(sub)
+        trans = Transaction.objects.get()
+        eq_(trans.status, constants.STATUS_FAILED)
+        eq_(trans.status_reason, 'settlement_declined wat')
+
+    def test_gateway_rejected(self):
+        sub = subscription(transactions=[
+            transaction(status='gateway_rejected',
+                        gateway_rejection_reason='wat'),
+        ])
+        self.process(sub)
+        trans = Transaction.objects.get()
+        eq_(trans.status, constants.STATUS_FAILED)
+        eq_(trans.status_reason, 'gateway_rejected wat')
+
+    def test_repeated(self):
+        sub = subscription(transactions=[transaction(status='settled')])
+        self.process(sub)
+        self.process(sub)
+        eq_(Transaction.objects.count(), 1)
+
+    def test_repeated_changed(self):
```

Comment:
Since there aren't any comments and this test name is vague, it took me a while to figure out what it's testing. Maybe: ``test_trans_status_must_not_change()`` ?